src/en/kunmangato/src/eu/kanade/tachiyomi/extension/en/kunmangato/KunMangaTo.kt

```diff
@@ -0,0 +1,267 @@
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.FormBody
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import uy.kohesive.injekt.injectLazy
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.Locale
+import kotlin.concurrent.thread
+
+class KunMangaTo : ParsedHttpSource() {
+    override val name = "KunMangaTo"
+    override val baseUrl = "https://kunmanga.to"
+    override val lang = "en"
+    override val supportsLatest = true
+
+    private val json: Json by injectLazy()
+
+    private val dateFormat by lazy {
+        SimpleDateFormat("MMM d, yyyy", Locale.ENGLISH)
+    }
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET(baseUrl)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = document.select(popularMangaSelector()).map { element ->
+            popularMangaFromElement(element)
+        }.associateBy({ it.url }, { it }).values.toList()
+        return MangasPage(mangas, false)
+    }
+
+    override fun popularMangaSelector(): String {
+        return ".sidebar-box-popular article"
+    }
+
+    override fun popularMangaFromElement(element: Element): SManga {
+        val a = element.select("a.manga").first()!!
+        return SManga.create().apply {
+            title = a.text()
+            url = a.attr("href").removePrefix(baseUrl)
+            thumbnail_url = element.select("img").first()!!.attr("src")
+        }
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val urlBuilder = "$baseUrl/search".toHttpUrl().newBuilder()
+            .addQueryParameter("term", query.trim())
+            .addQueryParameter("page", page.toString())
+
+        filters.filterIsInstance<UriPartFilter>().forEach { filter ->
+            urlBuilder.addQueryParameter(
+                filter.internalName,
+                filter.toUriPart(),
+            )
+        }
+
+        return GET(urlBuilder.build())
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = document.select(searchMangaSelector()).map { element ->
+            searchMangaFromElement(element)
+        }
+        val nextPageElement = document.select(".pagination .page-item").last()
```

Comment:
avoid the override if possible with the selector that I suggested

Reply:
Fixed